### ISSUE TITLE
JWK key with no 'use' field is causing validation of id_token failure

### ISSUE THREAD

**Issue by kelex** [damienbod/angular-auth-oidc-client#1079]
**Describe the bug**
Validation of id_token failed because JWK key has no `use` field.
I am pretty sure that Issue #440 is related.
Related #440 

**To Reproduce**
Having a server returning 
`/jwks` results with no `use` field
``` json
{
  "keys": [
    {
      "kty": "RSA",
      "n": "edited",
      "e": "edited"
    }
  ]
}
```
Cause the warning 
`angular-auth-oidc-client.js:211 no keys found, incorrect Signature, validation failed for id_token`
![image](https://user-images.githubusercontent.com/873284/116590978-a35fa700-a91e-11eb-88fe-79168159925c.png)

**Expected behavior**
The only available key is used, even when no 'use' field is provided.

cf. spec (https://openid.net/specs/openid-connect-discovery-1_0-final.html)

jwks_uri
>REQUIRED. URL of the OP's JSON Web Key Set [JWK] document. This contains the signing key(s) the RP uses to validate signatures from the OP.
>The JWK Set MAY also contain the Server's encryption key(s), which are used by RPs to encrypt requests to the Server. When both signing and encryption keys are made available, a use (Key Use) parameter value is REQUIRED for all keys in the referenced JWK Set to indicate each key's intended usage.
>**Although some algorithms allow the same key to be used for both signatures and encryption, doing so is NOT RECOMMENDED, as it is less secure**. The JWK x5c parameter MAY be used to provide X.509 representations of keys provided. When used, the bare key values MUST still be present and MUST match those in the certificate.

**Comment by andreaslarssen** [comment]
Experiencing the same issue. JWKS endpoint has one key, and therefore does not specify a "use" attribute.

Are there any temp. workarounds? This is a show stopper for me.

**Comment by kelex** [comment]
As workaround, for now I added an interceptor activated only on the `/jwks` URL that overload the http response body by adding the `use:sig` property when a single key is returned by the server.
This way, when the lib will properly handle this case, I will only have to remove this interceptor of the code base :)

I ll provide an example as soon as I come back on my work computer.

**Comment by andreaslarssen** [comment]
@kelex Cool. Did the same thing for now, thanks for the tip.

**Comment by andreaslarssen** [comment]
@kelex do you also use silentRenew? I have some issues with it

**Comment by damienbod** [comment]
thanks, released in version 11.6.9
